agent/agent_test.go

```diff
@@ -246,6 +246,63 @@ func TestAgent_ReconnectConfigSettings(t *testing.T) {
 	}()
 }
 
+func TestAgent_HTTPMaxHeaderBytes(t *testing.T) {
+	tests := []struct {
+		name                 string
+		maxHeaderBytes       int
+		expectError          bool
+		expectedHTTPResponse int
+	}{
+		{
+			"max header bytes 1 returns 431 http response when too large headers are sent",
+			1,
+			false,
+			431,
+		},
+		{
+			"max header bytes 0 returns 200 http response, as the http.DefaultMaxHeaderBytes size of 1MB is used",
+			0,
+			false,
+			200,
+		},
+		{
+			"negative maxHeaderBytes returns 200 http response, as the http.DefaultMaxHeaderBytes size of 1MB is used",
+			-10,
+			false,
+			200,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			a := NewTestAgent(t, fmt.Sprintf(`
+				http_config {
+					max_header_bytes = %d
+				}
+			`, tt.maxHeaderBytes))
```

Comment:
These test cases look good!

I know most of our tests use `TestAgent`, but in many cases that test helper does a lot more work than is necessary, which ends up wasting a lot of time in our test runs.

`TestAgent_ListenHTTP_MultipleAddresses` has an example of how `listenHTTP` can be tested without using `TestAgent`. I think that pattern would work here as well, and the tests should run a bit faster.

Reply:
It took a bit of work, but I got this functional without using NewTestAgent. 👍